pennylane/math/matrix_manipulation.py

```diff
@@ -165,50 +165,36 @@ def expand_and_reduce(op1_tuple: Tuple[np.ndarray, Wires], op2_tuple: Tuple[np.n
         mat1, wires1 = op1_tuple
         mat2, wires2 = op2_tuple
         expanded_wires = wires1 + wires2
-        mat1 = qml.math.expand_matrix(mat1, wires1, wire_order=expanded_wires)
-        mat2 = qml.math.expand_matrix(mat2, wires2, wire_order=expanded_wires)
+        mat1 = expand_matrix(mat1, wires1, wire_order=expanded_wires)
+        mat2 = expand_matrix(mat2, wires2, wire_order=expanded_wires)
         return reduce_func(mat1, mat2), expanded_wires
 
     reduced_mat, final_wires = reduce(expand_and_reduce, mats_and_wires_gen)
 
     return reduced_mat, final_wires
 
 
-def _local_sparse_swap_mat(i, n, format="csr"):
-    """Helper function which generates the sparse matrix of SWAP
-    for qubits: i <--> i+1 with final shape (2**n, 2**n)."""
-    assert i < n - 1
-    swap_mat = csr_matrix([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
-
-    j = i + 1  # i is the index of the qubit, j is the number of qubits prior to and include qubit i
-    return kron(
-        kron(eye(2 ** (j - 1)), swap_mat), eye(2 ** (n - (j + 1))), format=format
-    )  # (j - 1) + 2 + (n - (j+1)) = n
-
-
-def _sparse_swap_mat(i, j, n, format="csr"):
+def _sparse_swap_mat(qubit_i, qubit_j, n):
     """Helper function which generates the sparse matrix of SWAP
     for qubits: i <--> j with final shape (2**n, 2**n)."""
-    assert i < n and j < n
-    if i == j:
-        return eye(2**n, format=format)
-
-    (small_i, big_j) = (i, j) if i < j else (j, i)
-    store_swaps = [
-        _local_sparse_swap_mat(index, n, format=format) for index in range(small_i, big_j)
-    ]
-
-    res = eye(2**n, format=format)
-    for mat in store_swaps:  # swap i --> j
-        res @= mat
 
-    for mat in store_swaps[-2::-1]:  # bring j --> old_i
-        res @= mat
+    def swap_qubits(index, i, j):
+        s = list(format(index, f"0{n}b"))  # convert to binary
+        si, sj = s[i], s[j]
+        if si == sj:
+            return index
+        s[i], s[j] = sj, si  # swap qubits
+        return int(f"0b{''.join(s)}", 2)  # convert to int
 
-    return res
+    data = [1] * (2**n)
+    index_i = list(range(2**n))  # bras (we don't change anything)
+    index_j = [
+        swap_qubits(idx, qubit_i, qubit_j) for idx in index_i
+    ]  # kets (we swap qubits i and j): <10| --> <01|
+    return coo_matrix((data, (index_i, index_j)))
```

Comment:
Since the matrix generated here is intended for matrix multiplication, I believe the `csr_matrix` will show to be more performant.